src/sinks/humio_logs.rs

```diff
@@ -118,3 +118,156 @@ mod tests {
         assert_eq!((hec_event.time * 1000f64).fract(), 0f64);
     }
 }
+
+#[cfg(test)]
+#[cfg(feature = "humio-integration-tests")]
+mod integration_tests {
+    use super::*;
+    use crate::{
+        sinks::util::Compression,
+        test_util::{random_string, runtime},
+        topology::config::{SinkConfig, SinkContext},
+        Event,
+    };
+    use futures::compat::Future01CompatExt;
+    use futures01::Sink;
+    use serde_json::json;
+    use serde_json::Value as JsonValue;
+
+    // matches humio container address
+    const HOST: &str = "http://localhost:8080";
+
+    #[test]
+    fn humio_insert_message() {
+        let mut rt = runtime();
+        let cx = SinkContext::new_test();
+
+        rt.block_on_std(async move {
+            let repo = create_repository().await;
+
+            let config = config(&repo.default_ingest_token);
+
+            let (sink, _) = config.build(cx).unwrap();
+
+            let message = random_string(100);
+            let event = Event::from(message.clone());
+
+            sink.send(event).compat().await.unwrap();
+
+            let entry = find_entry(repo.name.as_str(), message.as_str()).await;
+
+            assert_eq!(message, entry["message"].as_str().unwrap());
+            assert!(
+                entry.get("@error").is_none(),
+                "Humio encountered an error parsing this message: {}",
+                entry
+                    .get("@error_msg")
+                    .unwrap_or(&json!("no error message"))
+            );
+        });
+    }
+
+    #[derive(Debug)]
+    struct HumioRepository {
+        name: String,
+        default_ingest_token: String,
+    }
+
+    /// create a new test config with the given ingest token
+    fn config(token: &str) -> super::HumioLogsConfig {
+        super::HumioLogsConfig {
+            host: Some(HOST.to_string()),
+            token: token.to_string(),
+            compression: Compression::None,
+            encoding: Encoding::Json.into(),
+            batch: BatchConfig {
+                max_events: Some(1),
+                ..Default::default()
+            },
+            ..Default::default()
+        }
+    }
+
+    /// create a new test humio repository to publish to
+    async fn create_repository() -> HumioRepository {
+        let client = reqwest::Client::builder().build().unwrap();
+
+        // https://docs.humio.com/api/graphql/
+        let graphql_url = format!("{}/graphql", HOST);
```

Comment:
I took a brief look at https://github.com/graphql-rust/graphql-client but it didn't seem worth it to pull in that dependency for one request. We could consider it if we end up integrating with more GraphQL endpoints.

Also, is there a preferred way for building up URLs like this?

Reply:
It is acceptable to use extra dependencies for tests as @fanatid indicated. For issuing just one query we might question bringing in a larger crate, though. This one's a judgment call, and it looks like it's not worth it at this point.